docs/content/docs/concepts/glossary.md

```diff
@@ -25,182 +25,605 @@ under the License.
-output data streams or data sets. A transformation might change a data stream or data set on a
-per-record basis, but might also only change its partitioning or perform an aggregation. While
-[Operators](#operator) and [Functions](#function) are the "physical" parts of Flink's API,
-Transformations are only an API concept. Specifically, most transformations are
-implemented by certain [Operators](#operator).
+A transformation is applied to one or more [DataStreams](#datastreams) and results in one or more 
+output DataStreams. A transformation might change a DataStream on a [per-record](#record) basis, but 
+might also only change its [partitioning](#partition) or perform an [aggregation](#aggregation). While 
+[operators](#operator) and [functions](#function) are the "physical" parts of Flink's API, transformations 
+are an API concept. Specifically, most transformations are implemented by certain [operators](#operator).
+
+#### Tumbling Window
+
+This is a [window](#window) that groups elements by a specified window size. Tumbling windows have a 
+fixed size and do not overlap. For example, if you specify a tumbling window with a size of 5 minutes, 
+the current window will be evaluated and a new window will be started every five minutes.
+
+#### Tuple
+
+A composite data type that has a finite ordered list of immutable elements. 
+
+#### Unbounded streams
+
+Unbounded [DataStreams](#datastream) have a start but no defined end. They do not terminate, provide 
+data as it is generated, and must be continuously processed. 
+
+#### (User-Defined) Functions
+
+Functions are implemented by the user and encapsulate the application logic of a [Flink application](#(flink)-application). 
+Most functions are wrapped by a corresponding [operator](#operator).
+
+#### User-Defined Aggregate Function (UDAF)
+
+This type of user-defined function aggregates multiple values into a single value.
+
+#### User-Defined Scalar Function (UDSF)
+
+This type of user-defined function maps zero, one, or more [scalar](#scalar) values to a new scalar value.
+
+#### User-Defined Table-valued Function (UDTF)
+
+This type of user-defined function uses zero, one, or multiple [scalar](#scalar) values as input parameters 
+(including variable-length parameters). A UDTF returns any number of rows, rather than a single value. 
+The returned rows can consist of one or more columns.
+
+#### ValueState<T>
+
+This is a type of [keyed state](#keyed-state) where Flink will store a single object for each key.
+ValueState keeps a value that can be updated and retrieved (scoped to key of the input element so there 
+will possibly be one value for each key that the operation sees). The value can be set using update(T) 
+and retrieved using T value().
+
+#### Watermark
+
+This is the mechanism in Flink to measure progress in event time. Watermarks are special timestamped 
+elements that get inserted by watermark generators into a [DataStream](#datastream). They flow as part 
```

Comment:
stream

Reply:
What is the difference between a stream and a DataStream?

Reply:
stream is the concept, `DataStream` is the class in the DataStream APIs representing that concept